tests/test_java.py

```diff
@@ -718,3 +720,284 @@ def test_java_frames_include_no_semicolons(
+    application_pid: int,
+    assert_collapsed: AssertInCollapsed,
+    expected_flags: List[Dict[str, Union[str, List[str]]]],
+) -> None:
+    with make_java_profiler(
+        frequency=99,
+        storage_dir=str(tmp_path),
+        java_async_profiler_mode="cpu",
+    ) as profiler:
+        assert_jvm_flags_equal(
+            profiler._metadata.get_jvm_flags_serialized(psutil.Process(application_pid)), expected_flags
+        )
+
+
+@pytest.mark.parametrize(
+    "java_cli_flags,java_env_flags",
+    [
+        (
+            "-XX:SelfDestructTimer=5",
+            "-XX:+PrintCodeCache",
+        ),
+    ],
+)
+@pytest.mark.parametrize("in_container", [True])
+@pytest.mark.parametrize(
+    "application_image_tag,expected_flags",
+    [
+        (
+            "hotspot-jdk-8",
+            [
+                {
+                    "name": "PrintCodeCache",
+                    "type": "bool",
+                    "value": "true",
+                    "origin": "non-default",
+                    "kind": ["product"],
+                },
+                {
+                    "name": "SelfDestructTimer",
+                    "type": "intx",
+                    "value": "5",
+                    "origin": "non-default",
+                    "kind": ["product"],
+                },
+            ],
+        ),
+        (
+            "hotspot-jdk-11",
+            [
+                {
+                    "name": "PrintCodeCache",
+                    "type": "bool",
+                    "value": "true",
+                    "origin": "environment",
+                    "kind": ["product"],
+                },
+                {
+                    "name": "SelfDestructTimer",
+                    "type": "intx",
+                    "value": "5",
+                    "origin": "command line",
+                    "kind": ["product"],
+                },
+            ],
+        ),
+        (
+            "zing",
+            [
+                {
+                    "name": "PrintCodeCache",
+                    "type": "bool",
+                    "value": "true",
+                    "origin": "environment",
+                    "kind": ["product"],
+                },
+                {
+                    "name": "SelfDestructTimer",
+                    "type": "intx",
+                    "value": "5",
+                    "origin": "command line",
+                    "kind": ["product"],
+                },
+            ],
+        ),
+    ],
+)
+def test_get_cmdline_and_env_jvm_flags(
+    tmp_path: Path,
+    docker_client: DockerClient,
+    application_docker_image: Image,
+    assert_collapsed: AssertInCollapsed,
+    java_cli_flags: str,
+    java_env_flags: str,
+    expected_flags: List[Dict[str, Union[str, List[str]]]],
+    application_pid: int,
+    application_image_tag: str,
+) -> None:
+    """
+    Tests that we can profile a Java app that runs with non-java "comm", by reading the argv0 instead.
```

Comment:
Fix docstring to reflect what this test actually tests - 
1. Tests retrieving specific flags only
2. Tests retrival from env & commandline sources and reporting the correct source